lib/Tags/Tags.js

```diff
@@ -42,100 +47,117 @@ export default class Tags extends React.Component {
         records: PropTypes.arrayOf(PropTypes.object),
       }),
     }).isRequired,
-    stripes: PropTypes.shape({
-      connect: PropTypes.func.isRequired,
-    }),
   };
 
-  constructor(props) {
-    super(props);
-    this.onAdd = this.onAdd.bind(this);
-    this.onRemove = this.onRemove.bind(this);
-    this.handleRemove = this.handleRemove.bind(this);
-    this.calloutRef = React.createRef();
+  static defaultProps = { onToggle: noop };
+
+  componentDidUpdate(prevProps) {
+    // TODO: remove the explicit `refreshRemote` call when https://issues.folio.org/browse/STCON-81 is done.
+    // Calling `refreshRemote` is needed to update entities resource when link prop is changed.
+    // `stripes-connect` is supposed to do it under the hood but it is a bug, and therefore it is needed to do it manually.
+    if (prevProps.link !== this.props.link) {
+      this.props.refreshRemote(this.props);
+    }
   }
 
-  onAdd(tags) {
+  calloutRef = React.createRef();
+
+  onAdd = tags => {
     this.saveEntityTags(tags);
     this.saveTags(tags);
-  }
+  };
 
-  onRemove(tag) {
+  onRemove = tag => {
     const entity = this.getEntity();
-    const tags = (entity.tags || {}).tagList || [];
+    const tags = this.getEntityTags();
     const tagList = tags.filter(t => t !== tag);
     entity.tags = { tagList };
     this.props.mutator.entities.PUT(entity);
-  }
+  };
 
   // add tag to the list of entity tags
   saveEntityTags(tags) {
     const entity = this.getEntity();
-    const tagList = (entity.tags || {}).tagList || [];
+    const tagList = this.getEntityTags();
     entity.tags = { tagList: sortBy(uniq([...tags, ...tagList])) };
     this.props.mutator.entities.PUT(entity);
   }
 
   // add tags to global list of tags
   saveTags(tags) {
-    const { mutator, resources } = this.props;
-    const records = (resources.tags || {}).records || [];
+    const records = this.getTags();
     const newTag = difference(tags, records.map(t => t.label.toLowerCase()));
 
     if (!newTag || !newTag.length) return;
 
-    mutator.tags.POST({
+    this.props.mutator.tags.POST({
       label: newTag[0],
       description: newTag[0],
     });
 
-    if (this.callout) {
+    if (this.calloutRef.current) {
       const message = <FormattedMessage id="stripes-smart-components.newTagCreated" />;
       this.calloutRef.current.sendCallout({ message });
     }
   }
 
-  handleRemove(tag) {
-    const entity = this.getEntity();
-    const tags = this.getEntityTags();
-    const tagList = tags.filter(t => (t !== tag));
-    entity.tags = { tagList };
-    this.props.mutator.entities.PUT(entity);
-  }
-
   getEntity() {
-    const entities = (this.props.resources.entities || {}).records || [];
-    return entities[0] || {};
+    return get(this.props, ['resources', 'entities', 'records', 0], {});
```

Comment:
I guess `resources` will always be present so so you can do
`get(this.props.resources, ['entities', 'records', 0], {})`

Reply:
Thank you, but I would prefer to leave it as is, just in case.